### PR TITLE
v2.x dynamic add_procs

### PR DESCRIPTION
**Pull request by hjelmn**
This should be all of the commits associated with the new add_procs behavior. While this is being tested I will double-check that nothing has been missed.

:bot:label:enhancement
:bot:assign: @jsquyres 
:bot:milestone:v2.0.0

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hjelmn** [comment]
This includes all the changes for oshmem from @igor-ivanov and portals4 from @tkordenbrock.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/782/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/783/ for details.

**Comment by hjelmn** [comment]
:bot:retest:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/786/ for details.

File: ompi/communicator/comm.c [hunk 1]
```diff
@@ -170,6 +178,8 @@ int ompi_comm_set_nb ( ompi_communicator_t **ncomm,
 
     /* Set remote group and duplicate the local comm, if applicable */
     if (0 < remote_size) {
+        ompi_communicator_t *old_localcomm;
```
Comment (jsquyres) [inline ompi/communicator/comm.c]: AFAICT, this variable isn't used.

Comment (jsquyres) [inline ompi/communicator/comm.c]: Disregard - the block got longer, and it's used below...

**Comment by igor-ivanov** [comment]
@hjelmn I would like to add following commits too
https://github.com/igor-ivanov/ompi/commit/9f12098ab8ec2e936a00b34be14ea6be0f199509
https://github.com/igor-ivanov/ompi/commit/69c82df781aaec587067839beceab29cc982d583
they are done as part of https://github.com/open-mpi/ompi/pull/908

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/807/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/842/ for details.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
see if the IU jenkins is happier now.

bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/857/ for details.

**Comment by jsquyres** [comment]
@hjelmn Still looks like there's some kind of segv in OSHMEM.

**Comment by hjelmn** [comment]
:bot:label:blocker

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/873/ for details.

**Comment by jsquyres** [comment]
:+1:

**Comment by jsquyres** [comment]
@hppritcha The PR is holding up a few others, so I'd like to merge it in and see how MTT does with it.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/878/ for details.

**Pull request merged**